### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by ahmed-alfarouq**
What I did:
- Refactor the Calculator component `components/Calculator.js` from a class-based component to a functional component with hooks
- Refactor the App component `App.js` from a class-based component to a functional component with hooks

### CODE REVIEW

**Review by henatan99 — COMMENTED** [review]
Hi @ahmed-al-farouq , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has conflicting files and the Github linters check could not run. 
image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.js [hunk 1]
```diff
@@ -1,42 +1,30 @@
...
-      },
-    };
-  }
-
-  updateState = (newData) => {
-    this.setState({
-      data: newData,
-    });
-  }
+export default function App() {
```
Comment (Davidosky007) [inline src/App.js:5]: * Kindly fix this error in the console as i get this error whenever i run `npm start` 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @ahmed-al-farouq 👋, 

Good job so far! 💪 
* Descriptive commit messages ✔️ 
* No linter errors ✔️ 
* Well written PR ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (ahmed-alfarouq) [reply src/App.js:5]: Hi @Davidosky007,
This error doesn't appear on my machine. So can you kindly tell me what should I do?
Thanks for the feedback

File: src/components/Calculator.jsx [hunk 2]
```diff
@@ -1,46 +1,41 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import calculate from '../logic/calculate';
 
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends Component {
-  updateData = (e) => {
-    const { data, updateState } = this.props;
+function Calculator({ data, value, updateState }) {
```
Comment (rukundoeric) [inline src/components/Calculator.jsx:5]: - [ ] <b>OPTIONAL:</b>  Prefer using ES6 arrow function. Actually, there is nothing wrong with ES5 functions syntax but it is mandatory to use es6 features in the javascript module. besides, Arrow functions provide a simplified structure, allowing for a more concise way to write function expressions. which makes them better than ES5 functions(function declarations). Here is the syntax of an ES6 arrow function...
```javascript
  const functionName = () => {
     // code here
  }
```

**Review by rukundoeric — APPROVED** [review]
Hi @ahmed-al-farouq,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Check the comments inder review


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Calculator.jsx (hunk 2 above)
Comment (ahmed-alfarouq) [reply src/components/Calculator.jsx:5]: Thanks for the feedback. I will work on that.

**Pull request merged**